atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ProfilingKeyValueService.java

```diff
@@ -132,6 +133,19 @@ private ProfilingKeyValueService(KeyValueService delegate) {
                         LoggingArgs.durationMillis(stopwatch));
     }
 
+    private static BiConsumer<LoggingFunction, Stopwatch> logTimeAndTableRows(String method,
+            TableReference tableRef,
+            Iterable<byte[]> rows) {
+        long startTime = System.currentTimeMillis();
+        return (logger, stopwatch) ->
+                logger.log("Call to KVS.{} at time {}, on table {} for (estimated) {} rows took {} ms.",
+                        LoggingArgs.method(method),
+                        LoggingArgs.startTimeMillis(startTime),
+                        LoggingArgs.tableRef(tableRef),
+                        LoggingArgs.rowCount(Ints.saturatedCast(rows.spliterator().estimateSize())),
```

Comment:
Glad we are coding defensively to not mess up our logging if deleting more than `Integer.MAX_VALUE` takes longer than a second 🗡

Reply:
Yeah... I think it's a defensive habit, but I end up thinking twice when I downcast things